### PR TITLE
added tests with switch

### PR DESCRIPTION
**Pull request by Graur**
Closes: #172

What's done:
- added test with `switch` and `close-to` matcher
- redo `close-to` matcher by adding check for float number and remove unnecessary checks from `number.abs` object
- added slow test from original issue

Looks like now `switch` with `close-to` matcher works faster.

---

## PR-Codex overview
This PR updates the `eo-hamcrest` package and improves the `close-to` matcher to specifically handle floating point numbers.

### Detailed summary
- Updated `eo-hamcrest` package to version `0.29.3`
- Improved `close-to` matcher to handle floating point numbers
- Added tests and failed output messages for the new `close-to` matcher
- Other minor improvements in test files

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by Graur** [comment]
@yegor256 Can we merge it?

**Comment by Graur** [comment]
@rultor merge

**Comment by Graur** [comment]
@levBagryansky please, have a look

**Comment by Graur** [comment]
@levBagryansky wait a bit

**Comment by Graur** [comment]
@levBagryansky Now it looks better. Please, take a look

File: src/main/eo/org/eolang/hamcrest/matchers/close-to-matcher.eo [hunk 1]
```diff
@@ -34,15 +34,32 @@
...
+
     [] > actual-delta
       minus. > @
-        abs.
-          number (a.minus operand)
+        float-abs
+          a.minus operand
         err
 
+    [value] > float-abs
```
Comment (levBagryansky) [inline src/main/eo/org/eolang/hamcrest/matchers/close-to-matcher.eo:57]: @Graur Is the point of this object that it is faster than `(number value).abs`?

**Review by levBagryansky — COMMENTED** [review]
@Graur can you explain why now it is faster?

File: src/main/eo/org/eolang/hamcrest/matchers/close-to-matcher.eo (hunk 1 above)
Comment (Graur) [reply src/main/eo/org/eolang/hamcrest/matchers/close-to-matcher.eo:57]: @levBagryansky Yes, it is.

File: src/test/eo/org/eolang/hamcrest/close-to-tests.eo [hunk 2]
```diff
@@ -40,3 +40,26 @@
...
+  assert-that > @
+    switch
+      *
+        TRUE
+        1.0
+    $.close-to
+      1.0
+      0.1
+
+[] > slow-test-with-switch
```
Comment (levBagryansky) [inline src/test/eo/org/eolang/hamcrest/close-to-tests.eo:54]: Can we add this test:
```
[] > switch-in-switch-in-switch-close-to
  assert-that > @
    switch
      *
        TRUE
        switch
          *
            TRUE
            switch
              *
                TRUE
                1.0
    $.close-to
      1.0
      0.1
```

**Review by levBagryansky — COMMENTED** [review]
@Graur What about the test above? it is from https://github.com/objectionary/eo-hamcrest/pull/176

**Comment by Graur** [comment]
@levBagryansky Sorry for delay. I've added new test. Check please

**Review by levBagryansky — APPROVED** [review]
@Graur Looks good to me!. Thank you

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**